### PR TITLE
Added warnings for `optimize`

### PR DESCRIPTION
**Pull request by anirban-chaudhuri**

### CODE REVIEW

File: pyciemss/interfaces.py [hunk 1]
```diff
@@ -785,11 +788,11 @@ def optimize(
             - A callable function defining the quantity of interest to optimize over.
         risk_bounds: float
             - The threshold on the risk constraint.
-        static_parameter_interventions: Dict[torch.Tensor, str]
-            - A dictionary of the form {intervention_time: parameter_name}
-              of static parameter interventions to optimize over.
-            - Each key is the time at which the intervention is applied.
-            - Each value is a string with the intervention parameter name.
+        static_parameter_interventions: Callable[[torch.Tensor], Dict[float, Dict[str, Intervention]]]
```
Comment (SamWitty) [inline pyciemss/interfaces.py:791]: Could you move this into a separate PR. A good fix, but unrelated to adding warnings.

**Review by SamWitty — CHANGES_REQUESTED** [review]
minor change requested (plus fix to linter)

File: pyciemss/interfaces.py (hunk 1 above)
Comment (anirban-chaudhuri) [reply pyciemss/interfaces.py:791]: I have now added this to #544. You can merge that first and then this PR.

**Comment by SamWitty** [comment]
@anirban-chaudhuri , could you merge `main` into this branch? It's still showing the change to documentation here.

**Review by SamWitty — APPROVED** [review]
